clients/notification-kafka/src/main/java/org/eclipse/hono/client/notification/NotificationTopicHelper.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.client.notification;
+
+import org.eclipse.hono.client.kafka.HonoTopic;
+import org.eclipse.hono.notification.AbstractNotification;
+import org.eclipse.hono.notification.deviceregistry.CredentialsChangeNotification;
+import org.eclipse.hono.notification.deviceregistry.DeviceChangeNotification;
+import org.eclipse.hono.notification.deviceregistry.TenantChangeNotification;
+
+/**
+ * Utility methods to determine the Kafka topic for a given type of notifications.
+ */
+public final class NotificationTopicHelper {
+
+    private NotificationTopicHelper() {
+        // prevent instantiation
+    }
+
+    /**
+     * Gets the topic name to publish a notification at.
+     *
+     * @param notification The notification to determine the topic for.
+     * @return The topic name.
+     */
+    public static String getTopicName(final AbstractNotification notification) {
+        final String address = notification.getAddress();
+        return getTopicNameForAddress(address);
+    }
+
+    /**
+     * Gets the topic name to consume notifications from.
+     *
+     * @param notificationType The class of the notifications to consume.
+     * @param <T> The type of notifications to consume.
+     * @return The topic name.
+     * @throws IllegalArgumentException If the given type is not a known subclass of {@link AbstractNotification}.
+     */
+    public static <T extends AbstractNotification> String getTopicName(final Class<T> notificationType) {
+        final String address;
+        if (TenantChangeNotification.class.equals(notificationType)) {
+            address = TenantChangeNotification.ADDRESS;
+        } else if (DeviceChangeNotification.class.equals(notificationType)) {
+            address = DeviceChangeNotification.ADDRESS;
+        } else if (CredentialsChangeNotification.class.equals(notificationType)) {
+            address = CredentialsChangeNotification.ADDRESS;
+        } else {
+            throw new IllegalArgumentException("Unknown notification type " + notificationType.getName());
+        }
+
+        return getTopicNameForAddress(address);
+    }
```

Comment:
Is there a reason why we need both these methods? Can't we always derive the address from the `ADDRESS` field?

Reply:
> Can't we always derive the address from the `ADDRESS` field?

That is exactly what the methods do. The address is always simply a constant defined by the ADDRESS field. The topic name is the address prefixed by `hono.notification.`. On the receiver side, we need to determine which address field to use. The point here is that we need a static value because the client subscribes to it before having an instance of the notification at hand. 
We could move the logic into the Kafka sender respectively receiver implementations but I don't know any benefits of this. FMPOV it is clearer to have one place where the logic resides that maps between notification types and their topics.

Reply:
Ah... Now I get it. I'll do this.